code/modules/food_and_drinks/pizzabox.dm

```diff
@@ -200,6 +225,34 @@
 			unprocess()
 	return
 
+/obj/item/pizzabox/attack(mob/living/target, mob/living/user, def_zone)
+	. = ..()
+	if(boxes.len >= 3 && prob(25 * boxes.len))
+		disperse_pizzas()
+
+/obj/item/pizzabox/throw_impact(atom/movable/AM)
+	if(boxes.len >= 2 && prob(20 * boxes.len))
+		disperse_pizzas()
+
+/obj/item/pizzabox/proc/disperse_pizzas()
+	visible_message("<span class='warning'>The pizzas fall everywhere!</span>")
+	for(var/V in boxes)
+		var/obj/item/pizzabox/P = V
+		var/fall_dir = pick(GLOB.alldirs)
+		step(P, fall_dir)

```

Comment:
no need, step doesn't go through walls, the box just won't move if it tries to step onto a wall/something dense.